http-proxy-policy-operator/src/policy.py

```diff
@@ -0,0 +1,270 @@
+        password: The password.
+
+    Raises:
+        RuntimeError: If the action failed.
+    """
+    script = textwrap.dedent(
+        f"""\
+        from django.contrib.auth import get_user_model
+        User = get_user_model()
+        username, password = {repr(username)}, {repr(password)}
+        user, created = User.objects.get_or_create(username=username)
+        if created:
+            user.set_password(password)
+            user.save()
+        elif not user.check_password(password):
+            user.set_password(password)
+            user.save()
+        user.is_staff = True
+        user.is_superuser = True
+        user.save()
+        """
+    )
+    try:
+        subprocess.run(  # nosec
+            ["charmed-http-proxy-policy.manage", "shell"],
+            check=True,
+            input=script,
+            encoding="utf-8",
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        )
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(f"failed to update Django user: {e.stdout}") from e
+
+
+class EvaluatedHttpProxyRequest(http_proxy.HttpProxyRequest):
+    """Response returned from the HTTP proxy policy refresh API.
+
+    Attributes:
+        status: HTTP proxy request status.
+        accepted_auth: accepted HTTP proxy authentication method.
+    """
+
+    status: str
+    accepted_auth: Optional[str]
+
+
+class HttpProxyPolicyClient:
+    """HTTP proxy policy client."""
+
+    def __init__(
+        self, username: str, password: str, endpoint: str = "http://localhost:8080"
+    ) -> None:
+        """Initialize the http proxy policy client.
+
+        Args:
+            endpoint: http proxy policy endpoint.
+            username: http proxy policy username.
+            password: http proxy policy password.
+        """
+        self._endpoint = endpoint
+        self._auth = (username, password)
+
+    def refresh(
+        self, proxy_requests: list[http_proxy.HttpProxyRequest]
+    ) -> list[EvaluatedHttpProxyRequest]:
+        """Refresh HTTP proxy requests.
+
+        This updates all requests stored in the HTTP proxy policy and re-evaluates the rules
+        against the updated requests. The evaluated requests are returned.
+
+        Args:
+            proxy_requests: list of http proxy requests.
+
+        Returns:
+            list of evaluated http proxy requests.
+        """
+        input_requests = {str(r.id): r for r in proxy_requests}
+        payload = []
+        for request in proxy_requests:
+            request_dict = request.model_dump(mode="json")
+            del request_dict["group"]
+            request_dict["requirer"] = request_dict["id"]
+            del request_dict["id"]
+            payload.append(request_dict)
+        response = requests.post(
+            self._endpoint + "/api/v1/requests/refresh",
+            json=payload,
+            auth=self._auth,
+            timeout=10,
+        )
+        response.raise_for_status()
```

Comment:
We might want to catch these exception and wrap them in a business error with an explicit message